### DISCUSSION TITLE
Can't get insertContent to insert DIV node

### DISCUSSION THREAD

**Discussion by piotrkulpinski** [ueberdosis/tiptap#3918] [General Questions]
Hi,

I'm using oEmbed to generate HTML code from provided URL. Once I have the html, I try to inject it into the editor via `insertContent` method, but I can't get it to work with DIV elements.

Here's the code:
```tsx
const addOembed = () => {
  const html =
    '<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;"><iframe src="https://player.vimeo.com/video/141567420?h=54a5f22794&app_id=122963&byline=0&badge=0&portrait=0&title=0" style="top: 0; left: 0; width: 100%; height: 100%; position: absolute; border: 0;" allowfullscreen scrolling="no" allow="encrypted-media;"></iframe>cze</div>'
  
  editor?.commands.insertContent(html)
}

return (
  <>
    <Button type="button" size="tiny" title="OEmbed" tabIndex={-1} onClick={addOembed}>
      <IconBrandYoutube className="button-tiny-icon" />
    </Button>
  </>
```

Does anyone know how to get it to insert the following content and display the embedded iframe? 

Thanks!

**Comment by codemzy** [comment]
Rather than inserting the html, you might want to consider [node views](https://tiptap.dev/guide/node-views). E.g. you insert `<vimeo src="https://player.vimeo.com/video/141567420">` and the node view handles injecting all the HTML. 

I built something like this for YouTube and Vimeo embeds - so this blog post might give you a few pointers: https://www.codemzy.com/blog/tiptap-video-embed-extension

**Reply by piotrkulpinski** [reply]
Thanks! This is really helpful. 

Do you have any idea how to allow the editor to insert raw HTML? I'm using a tool that's turning embed links from YouTube, Twitter etc, into embeddable HTML code. 

I then want to insert the code directly into the editor.

Tried something like this, but it doesn't work: 
```js
addNodeView() {
  return ({ editor, node }) => {
    const div = document.createElement('div')
    div.className = 'oembed-wrapper'
    div.innerHTML = node.attrs.html

    return {
      dom: div,
    }
  }
},
```

**Reply by codemzy** [reply]
I don't think you would want the full html as an attribute on your custom extension, it would bloat your saved HTML quite a bit. Better to add the repeating html in the node view, and have your parsed html for the custom node as something like `<oembed src="https://player.vimeo.com/video/141567420?h=54a5f22794"></oembed>` but - looking at oEmbed, I kinda think oEmbed and TipTap might conflict with each other because oEmbed is trying to transform the HTML on the page, and that's what TipTap is in control of parsing and rendering.

I'm guessing you will either need to allow all that extra html to be added and parsed (with your own custom extension) - but it will bloat your stored HTML if you are saving it, or use the [Tiptap YouTube extension](https://tiptap.dev/api/nodes/youtube), or something like the custom video extension I wrote about in that blog post, so you can keep your stored HTML lean and add the extra embed markup in the node view.

Hope that makes sense!
